pkg/security/probe/process_resolver.go

```diff
@@ -845,7 +845,7 @@ func (p *ProcessResolver) resolveFromProcfs(pid uint32, maxDepth int) *model.Pro
 	parent := p.resolveFromProcfs(ppid, maxDepth-1)
 	if inserted && entry != nil && parent != nil {
 		if parent.Equals(entry) {
-			entry.SetParent(parent)
+			entry.SetParentOfForkChild(parent)
```

Comment:
I wouldn't rename this function because the purpose of this function is to set the parent no matter the type of relationship there is between the entry and the parent even if we use it only here for a fork.

Reply:
Ah, I renamed it based on the [comment](https://github.com/DataDog/datadog-agent/blob/main/pkg/security/secl/model/process_cache_entry.go#L101) describing the function. It's only used https://github.com/DataDog/datadog-agent/blob/main/pkg/security/secl/model/process_cache_entry.go#L120 and https://github.com/DataDog/datadog-agent/blob/main/pkg/security/probe/process_resolver.go#L845. I'm not sure what situation would result in `parent.Equals(entry)`. Also, this func requires the entry to be a thread. Would SetAncestorOfThread() make more sense?

Reply:
For my point of view, the name of the function should represent what the function is doing and not how/where we call it. That said I'm fin with changing is name to `SetParentOfForkChild`

Reply:
I agree with that. It is also setting ```pc.IsThread = true``` which isn't appropriate for a generic process context SetParent() func.